docs/compile_guard.md

```diff
@@ -74,3 +74,6 @@ by using the `ASSERT_ON` define.
 
 Note that logic inside a compile guard cannot be used outside of a compile
 guard (otherwise a signal may not have a driver when the guard is disabled).
+
+To generate an `ifndef` instead of `ifdef`, pass the argument `invert=True` to
```

Comment:
I think `invert` might not be the best phrasing (e.g. confusion between `ifdef !X` vs `ifndef X`). Perhaps `undefined=True` might be better? Same for the related PRs.

Alternatively,  we could have a `type` argument, default to `type="defined"` and allow `type="undefined"`. TBH, I like this more for the explicitness (and flexibility, although I don't think Verilog conditional compilation allows for much more...); but, the single boolean flag is prob more ergonomic.